src/test/path-transformers_test.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+
+import {assert} from 'chai';
+import {join as pathJoin, sep as pathSeparator} from 'path';
+
+import {isPlatformWindows, pathFromUrl, urlFromPath} from '../path-transformers';
+
+const WindowsRootPath = 'C:\\Users\\TEST_USER\\TEST_ROOT';
+const MacRootPath = '/Users/TEST_USER/TEST_ROOT';
+const RootPath = isPlatformWindows() ? WindowsRootPath : MacRootPath;
+
+suite('pathFromUrl()', () => {
+
+  test('creates a filesystem path using the platform separators', () => {
+    const otherSeparator = pathSeparator === '/' ? '\\' : '/';
+    const path = pathFromUrl(RootPath, '/some/url/pathname');
+    assert.include(path, pathSeparator);
+    assert.notInclude(path, otherSeparator);
+  });
+
+  test('returns a path if url is absolute', () => {
+    const path = pathFromUrl(RootPath, '/absolute/path');
+    assert.equal(path, pathJoin(RootPath, 'absolute', 'path'));
```

Comment:
It kinda seems like this is duplicating work that the Analyzer's resolver and loader should handle. We should strive as much as possible to do url -> fs path conversion in configurable places as in the resolvers and loaders, as this is something that clients, especially big ones, need to customize.

For one example, `../paper-button/paper-button.html` in a default app needs to end up pointing to `rootDir/bower_components/paper-button/paper-button.html`

I don't have a global view of this, but it doesn't seem like we're maintaining all the info needed to do the right thing here.